src/topology/uniform_space/basic.lean

```diff
@@ -1128,6 +1128,31 @@ lemma uniform_space.comap_comap {α β γ} [uγ : uniform_space γ] {f : α →
   uniform_space.comap (g ∘ f) uγ = uniform_space.comap f (uniform_space.comap g uγ) :=
 by ext ; dsimp [uniform_space.comap] ; rw filter.comap_comap
 
+lemma uniform_space.comap_inf {α γ} {u₁ u₂ : uniform_space γ} {f : α → γ} :
+  (u₁ ⊓ u₂).comap f = u₁.comap f ⊓ u₂.comap f :=
+begin
+  ext : 1,
+  change (𝓤 _) = (𝓤 _),
```

Comment:
I think this indicates that the `ext` lemma is bad; it seems to be about the wrong `uniformity`

Reply:
Moving `uniform_space_eq` below `uniform_space.uniformity` would fix that.

Reply:
Yes I was thinking of making a refactor PR to make `uniformity` the "normal form" of `uniform_space.to_core.uniformity` by `uniformity` (or just dropping `uniformity` and assigning the notation to `uniform_space.to_core.uniformity`, I guess that would work too).
But I'd prefer doing that in its own PR, does that feel right ?

Reply:
A separate PR would indeed be a good idea; would you prefer to merge this as is then make that PR, or pause this until that PR is done?

I think making the notation be for `uniform_core.uniformity` is actually the way to go (and remove the existing def)

Reply:
I think I would prefer this to be merged as is, because #14534 is a big PR which will probably take some time to be reviewed, so I would like the reviewing to start as soon as possible, and currently it is blocked by this PR.